### DISCUSSION TITLE
使用java SDK操作collection、partition、index、data等，若出现异常时，能否将关键异常信息返回

### DISCUSSION THREAD

**Discussion by LiJie20190102** [milvus-io/milvus#23457] [Ideas & Feature requests]
下图是我进行矢量查询是，查询失败，但是未提示我任何错误原因。不止查询如此，当我操作collection、partition、index、data等时，若出现异常，好像也不提示异常信息。CLI会进行提示，有时还需要借助CLI来查看错误信息，我认为这是不可取的。
![image](https://user-images.githubusercontent.com/53458004/232232808-fadc2104-fef7-4e4d-a831-2a9d30dd43a8.png)

以上只是我初步使用Java SDK遇到的感受，也有可能是我使用不当导致我查看不到异常信息。同时，我不知道其他编程语言的SDK是否有异常信息提示。我若理解有误解，请及时回复我。

**Comment by yhmo** [comment]
每个API返回的都是一个包装过的对象： R<XXXX>
R是一个包装类：https://github.com/milvus-io/milvus-sdk-java/blob/2.2/src/main/java/io/milvus/param/R.java
如果API出错，这个R的status非零，并且通过getMessage可获得报错信息。

```
        if (respSearch.getStatus() != R.Status.Success.getCode()) {
            System.out.println("ERROR: "+respSearch.getMessage());
        }
```
